src/audio/smart_amp/smart_amp.c

```diff
@@ -517,6 +528,20 @@ static int smart_amp_trigger(struct comp_dev *dev, int cmd)
 	comp_dbg(dev, "smart_amp_trigger(), command = %u", cmd);
 
 	ret = comp_set_state(dev, cmd);
+	if (ret < 0)
+		return ret;
+
+	if (ret == COMP_STATUS_STATE_ALREADY_SET)
+		return PPL_STATUS_PATH_STOP;
+
+	/* don't stop smart_amp on pause or if the feedback source is active/paused */
+	if (cmd == COMP_TRIGGER_PAUSE ||
```

Comment:
this isn't fun to read, it looks like you can break this if up into a couple different checks with the same exit, please do.

Reply:
this is copy/pasted from the mixer, and I filed a separate issue to figure out what the behavior is supposed to be. 

see https://github.com/thesofproject/sof/issues/3836, this whole thing is a real mess.

Reply:
sounds like we are in a fun state regarding multi-input components

Reply:
yep. Fun as in 'confused'.

I honestly don't know why this smart-amp component needed to be special cased with a source and a 'feedback' source. They are both sources.
when we have stuff like AEC, are we also going to special case?

It seems like we are not making use of proper abstractions or 'object hiearchy'. Changes on the sources should be dealt with in the same manner, be it the second source of a mixer or a feedback. this feels like we are still going from a linear pipeline to a graph and haven't fully conceptualized what's missing.

Reply:
Do we have the ability to distinguish sources by anything other than index? if yes then I agree we should collapse them